README.md

```diff
@@ -1,7 +1,151 @@
+
+## Using object-store-client
+    
+### Add object-store-client dependency
+
+In the SBT build add:
+
+```sbt
+resolvers += Resolver.bintrayRepo("hmrc", "releases")
+
+libraryDependencies += "uk.gov.hmrc.objectstore" %% "object-store-client-play-xx" % "x.x.x"
+```
+
+### Configuration
+
+- Ensure `appName` is configured in `application.conf`
+
+
+- Enable `uk.gov.hmrc.objectstore.client.play.modules.ObjectStoreModule` in the `application.conf` by adding
+```hocon
+play.modules.enabled += "uk.gov.hmrc.objectstore.client.play.modules.ObjectStoreModule"
+```
+
+- Configure object-store HTTP configuration under `microservice.services` block in `application.conf`
+```hocon
+object-store {
+  host = localhost
+  port = 8464
+}
+```
+
+Also, override this configuration in app-config-base so this service could interact with object-store service in deployed environments. 
+
+- Configure Authorization token in `application.conf` which grants permissions to this service to interact with object-store via object-store-client.
+```hocon
+internal-auth.token = "<INSERT-VALUE>"
+```
+For the deployed environments, please request Build & Deployment team to generate a token for the service and configure this property with the encrypted value of the token.
+
+- Configure `object-store.default-retention-period` in `application.conf`. This property represents the default retention period of the uploaded object in object-store. 
+  This value can be overriden on the individual upload requests. 
+
+```hocon
+object-store.default-retention-period = "1-week"
+```
+
+The valid values for this configuration are 
+- 1-week
+- 1-month
+- 6-months
+- 1-year
+- 7-years
+- 10-years
+
+Any other configured value or absence of configuration will result in the failure of the service start up.
+
+### Available operations
+
+Use DI (Guice or your preferred mechanism) to inject an instance of `uk.gov.hmrc.objectstore.client.play.PlayObjectStoreClient` in the controllers or services
+and use its methods to interact with object-store. Here onwards, we'll call that instance as `client` in the code excerpts.
+
+
+#### **Put object**
+
+```scala
+import uk.gov.hmrc.objectstore.client.play.Implicits._
+
+client.putObject(
+  path = Path.Directory("accounts/2020").file("summary.txt"),
+  content = "this is the content",
+  retentionPeriod = RetentionPeriod.OneMonth, // defaults to 'object-store.default-retention-period' configuration
+  contentType = Some("plain/text"), // defaults to None
+  owner = "my-service" // defaults to 'appName' configuration
+) // returns Future[Unit]
+```
+
+The above code will put the value of `content` parameter in an object at the path `/my-service/accounts/2020/summary.text` with a retention period of 1 month.
+Bear in mind, that this operation will overwrite the content that already exist at that path.
+
+The method `putObject` is polymorphic in its parameter `content`. At the moment, we support following types -
```

Comment:
is it worth saying that we recommend using `Source[ByteString, _]` to avoid loading it all into memory. Or at least for large files?